src/jdk.compiler/share/classes/com/sun/tools/javac/tree/JCTree.java

```diff
@@ -623,6 +623,30 @@ public <R,D> R accept(TreeVisitor<R,D> v, D d) {
         public Tag getTag() {
             return TOPLEVEL;
         }
+
+        public boolean isAnonymousMainClass() {
```

Comment:
This method seems a bit confusing to me. I believe the fields and methods will be stripped from `defs` if/when the wrapping class is created, which will mean this method will start to return `false`, no? It overall does not seem like a generally useful predicate.

(If I understand this correctly, if we created the wrapping class in parser neither of these two methods would be needed.)

Reply:
Inlining